aptos-move/mvhashmap/src/versioned_data.rs

```diff
@@ -294,4 +269,28 @@ impl<K: Hash + Clone + Eq, V: TransactionWrite> VersionedData<K, V> {
             }
         }));
     }
+
+    // When a transaction is committed, this method can be called for its delta outputs to add
+    // a 'shortcut' to the corresponding materialized aggregator value, so any subsequent reads
+    // do not have to traverse below the index. It must be guaranteed by the caller that the
+    // data recorded below this index will not change after the call, and that the corresponding
+    // transaction has indeed produced a delta recorded at the given key.
+    // If the result is err, it means the base value hadn't been set.
+    pub(crate) fn commit_delta(&self, key: &K, txn_idx: TxnIndex) -> anyhow::Result<u128> {
+        let mut v = self.values.get_mut(key).expect("Path must exist");
+
+        // +1 makes sure we include the delta from txn_idx.
+        match v.read(txn_idx + 1) {
+            Ok(MVDataOutput::Resolved(value)) => {
+                v.versioned_map
+                    .get_mut(&txn_idx)
+                    .expect("Entry by the txn must exist to commit delta")
+                    .record_delta_shortcut(value);
+
+                Ok(value)
+            },
+            Err(MVDataError::Unresolved(_)) => bail!("No base value"),
```

Comment:
Why the case when `read()` returns `Unresolved(..)` cannot happen here? If we have the first txn which we commit then the delta will not have any shortcuts, and so returns an unresolved call to go to storage?